sdk/nodejs/cloudwatch/widgets_json.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+
+import { YAxis } from "./widgets_graph";
+
+// All the typings for the json blob we need to create for Dashboard.dashboardBody. We'll convert
+// the right in-memory auto-laid-out system we have into these.  These typings are purely so we do
+// the json conversion properly as it's very easy to screw up if using things like the 'any' type.
+
+export interface WidgetJson {

```

Comment:
I like that you moved all of this into its own file. May do something similar w the Swagger JSON typings.